### PR TITLE
update gitlabv2 to tri-state

### PR DESCRIPTION
**Pull request by 0x1**
### Description:
updating gitlab v2 detector to use tri-state verification and fixing some of the tests in v1 and v2

### Checklist:
* [ ] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/usage/install/#local-installation))?

### CODE REVIEW

File: pkg/detectors/gitlab/gitlab.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			resMatch = keyString[len(keyString)-1]
 		}
 
-		secret := detectors.Result{
+		s1 := detectors.Result{
```
Comment (rosecodym) [inline pkg/detectors/gitlab/gitlab.go:56]: Did you rename this for similarity to other detectors?

File: pkg/detectors/gitlab/gitlabv2_test.go [hunk 2]
```diff
@@ -1,115 +0,0 @@
...
-	"testing"
-	"time"
-
-	"github.com/kylelemons/godebug/pretty"
-	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
-	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
-	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
-)
-
-func TestGitlabv2_FromChunk(t *testing.T) {
```
Comment (rosecodym) [inline pkg/detectors/gitlab/gitlabv2_test.go:15]: Why did you delete this test?

File: pkg/detectors/gitlabv2/gitlab.go [hunk 3]
```diff
@@ -42,7 +46,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 			continue
 		}
 
-		secret := detectors.Result{
+		s1 := detectors.Result{
```
Comment (rosecodym) [inline pkg/detectors/gitlabv2/gitlab.go:49]: same question as v1 rename

File: pkg/detectors/gitlabv2/gitlab_test.go [hunk 4]
```diff
@@ -31,18 +32,19 @@ func TestGitlab_FromChunk(t *testing.T) {
...
+		wantErr             bool
+		wantVerificationErr bool
 	}{
 		{
 			name: "found, verified",
 			s:    Scanner{},
 			args: args{
 				ctx:    context.Background(),
-				data:   []byte(fmt.Sprintf("You can find a aws secret %s within", secret)),
+				data:   []byte(fmt.Sprintf("You can find a gitlab secret %s within", secret)),
```
Comment (rosecodym) [inline pkg/detectors/gitlabv2/gitlab_test.go:47]: ❤️

**Review by rosecodym — COMMENTED** [review]
my only blocking question is about the deleted test

File: pkg/detectors/gitlab/gitlab.go (hunk 1 above)
Comment (0x1) [reply pkg/detectors/gitlab/gitlab.go:56]: yes, any detectors that i updated

File: pkg/detectors/gitlabv2/gitlab.go (hunk 3 above)
Comment (0x1) [reply pkg/detectors/gitlabv2/gitlab.go:49]: yes, i updated this for any detectors i touched

File: pkg/detectors/gitlab/gitlabv2_test.go (hunk 2 above)
Comment (0x1) [reply pkg/detectors/gitlab/gitlabv2_test.go:15]: because it was testing v2. i figured we should just have a test for v1 and a test for v2, not both. but i'm open to thoughts on if we should keep it

File: pkg/detectors/gitlabv2/gitlab.go (hunk 3 above)
Comment (0x1) [reply pkg/detectors/gitlabv2/gitlab.go:49]: there's a few other detectors that still use `secret`. i was planning on updating them after all the tristate detectors have been updated (if they don't get updated during the process).

**Comment by rosecodym** [comment]
per out-of-band conversation: this change is good, but we suspect it might not resolve the underlying flip-flopping issue. @0x1 is going to do some research to try to settle that question.

File: pkg/detectors/gitlab/gitlabv2_test.go (hunk 2 above)
Comment (rosecodym) [reply pkg/detectors/gitlab/gitlabv2_test.go:15]: oh, i totally missed that it was a v2 test in the v1 directory. now that i see that, i completely agree.

**Comment by 0x1** [comment]
we will be consolidating the gitlab detectors instead of adding tri-state to v2. PR here: https://github.com/trufflesecurity/trufflehog/pull/1776
